test/test_strerror.cpp

```diff
@@ -43,4 +43,12 @@ TEST(test_strerror, get_error) {
 
   rcutils_strerror(error_string, sizeof(error_string));
   ASSERT_STREQ(error_string, "No such file or directory");
+
+#if (!defined(_WIN32)) && (!( \
+    defined(_GNU_SOURCE) && (!defined(ANDROID) || __ANDROID_API__ >= 23)))
+  // Hopefully this does not become a valid errno.
+  errno = 12345;
+  rcutils_strerror(error_string, sizeof(error_string));
+  ASSERT_STREQ(error_string, "Failed to get error");
```

Comment:
@brawner a message to hint our future selves in case of failure would be nice.

Reply:
Added a message.